### DISCUSSION TITLE
How to set the valid time of presignurl to permanent

### DISCUSSION THREAD

**Discussion by novel045** [minio/minio#12754] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by Phynecs** [comment, marked as answer]
s3 specifies a max time for signed url of 7 days, see "X-Amz-Expires" in the official documentation: https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html

I don't think departing from the spec is a good idea.

There is an alternative way to do it, make a separate bucket where files are accessible without signed url, so a public bucket.
But be aware of the security considerations of this option.
